### PR TITLE
Add Goreleaser to build releases

### PR DESCRIPTION
**Pull request by dehanj**
The files needed to build the packages I have put in the `system/` folder. Some of these are duplicates from the `debian` folder, this is intentional since I think we can remove the ´Debian`folder going forward.

The only noticeable difference between our "old" Debian package and these are that the `tkey-ssh-agent.1` was gzip. It is not here.

Needs to be verified:
- [x] Ubuntu (deb-pkg)
- [x] Alpine installation 
- [x] Arch Linux
- [x] RPM
- [x] Windows binaries
- [x] macOS signature
- [x] macOS binaries

### CODE REVIEW

**Comment by dehanj** [comment]
The generated windows binaries are yet to be verified. But it is ready for review and testing.

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    # Formats to be generated.
+    formats:
+      - apk
+      - deb
+      - rpm
+      - archlinux # Since: v1.13
+
+    dependencies:
+      - pinentry-gnome3 | pinentry
```
Comment (dehanj) [inline .goreleaser.yaml:165]: Alpine does not support "|" to specify dependencies - seems like only Debian and derivatives does. 

It does not have "Pinentry-gnome3" either, but seems to have "pinentry-gnome". 
Probably needs to make an override for Alpine here.

Comment (dehanj) [reply .goreleaser.yaml:165]: Fixed, overrides dependencies for Alpine.

**Comment by dehanj** [comment]
I have continued to verify:
- Signature on our Darwin binary seems OK.
- Alpine is possible to install. Since we don't sign the packages at the moment one needs to pass `--allow-untrusted`. 
- We are shipping unnecessary files for alpine, most likely.

**Comment by dehanj** [comment]
Windows binaries (including tray) checked on windows.

**Comment by dehanj** [comment]
Wrapped the pre-hook command in "sh -c <cmd>" is the answer to make it run properly. Added a commit that fixes it.

**Review by mchack-work — APPROVED** [review]
LGTM.

**Pull request merged**